src/trans/gpu/internal/ext_acc.F90

```diff
@@ -0,0 +1,387 @@
+! (C) Copyright 2022- NVIDIA.
+!
+! This software is licensed under the terms of the Apache Licence Version 2.0
+! which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+! In applying this licence, ECMWF does not waive the privileges and immunities
+! granted to it by virtue of its status as an intergovernmental organisation
+! nor does it submit to any jurisdiction.
+module openacc_ext_type
```

Comment:
This new module is a hacky extra module that I had to add to make it work with the full IFS code. The problem is that when we call ectrans from IFS, it might happen that e.g. PGP2 and PGP3A partially overlap (like they are different slices of the same array). We can't move those arrays to OpenACC because after PGP2 is copies, PGP3A is partially present, and vice versa. This routines take care of that and copy the merged PGP2+PGP3A.